pkg/upgrader/upgrade/upgrade.go

```diff
@@ -47,6 +47,7 @@ const (
 func Upgrade(s *state.State) error {
 	// commonSteps are same for all worker nodes and they are safe to be run in parallel
 	commonSteps := []task.Task{
+		{Fn: state.GenerateBootstrapToken, ErrMsg: "failed to generate bootstrap token"},
```

Comment:
Why do we need this? I don't remember we had any problem requiring us a new token.

Reply:
Because inside pkg/templates/kubeadm we call for `kubeadmv1beta1.NewBootstrapTokenString(s.JoinToken)` (or `kubeadmv1beta2.NewBootstrapTokenString(s.JoinToken)`) which fails on case when token is empty, even that it's not required for upgrade procedures.